### DISCUSSION TITLE
HOWTO: Debug within Teams mobile app on iOS?

### DISCUSSION THREAD

**Discussion by VanessaRussell** [OfficeDev/microsoft-365-agents-toolkit#10218] [Q&A]
I have a Teams Tab app that successfully works within Teams Web and Teams Desktop; however, for an unknown reason it is not working correctly within Teams Mobile (iOS). Where do I even start on trying to debug my app within the Teams Mobile app?

I found this Learn Article: https://learn.microsoft.com/en-us/microsoftteams/platform/toolkit/debug-mobile?tabs=vscode%2Cios1%2Cios2#test-your-tab-app-on-mobile-client but, it is very light on details beyond ensuring that the app data is reset.

Am I overlooking a published guide somewhere other than that Learn Article?

**Comment by MuyangAmigo** [comment]
Hey @VanessaRussell Do you see any errors while running your app in iOS? Also looping @XiaofuHuang to help further.

**Comment by davrous** [comment]
Hi @VanessaRussell, if the app doesn't work on iOS, we have unfortunately no native way to debug it (attach to the iPhone using a USB cable from a Mac for instance). The only way you have is to use a tool like Vorlon.js: https://www.vorlonjs.com/, you can check some videos I did demonstrating how to debug the app running on iOS with Vorlon.js such as: https://youtu.be/77u73mD9MKE?si=wIrlyEY7m9yIHVh3&t=2789

**Comment by XiaofuHuang** [comment]
You can reset app data in iOS using two methods:
1. Reinstall the App
1. Clean App Data in iOS Settings. For more details, you can refer to the [Wiki document](https://github.com/OfficeDev/TeamsFx/wiki/Run-and-debug-your-Teams-application-on-iOS-or-Android-client) with a screenshot.

**Reply by VanessaRussell** [reply]
Thanks. The Learn article was helpful in successfully doing this. Unfortunately, that doesn't clear the issue. Therefore, I need some way to be able to delve into what is different about the iOS Teams environment versus the Web / Desktop Teams environment.
